graph/bidirectional_dijkstra.cpp

```diff
@@ -0,0 +1,268 @@
+/**
+ * @file
+ * @brief [Bidirectional Dijkstra Shortest Path Algorithm]
+ * (https://www.coursera.org/learn/algorithms-on-graphs/lecture/7ml18/bidirectional-dijkstra)
+ *
+ * @author [Marinovksy](http://github.com/Marinovsky)
+ *
+ * @details
+ * This is basically the same Dijkstra Algorithm but faster because it goes from
+ * the source to the target and from target to the source and stops when
+ * finding a vertex visited already by the direct search or the reverse one.
+ * Here some simulations of it:
+ * https://www.youtube.com/watch?v=DINCL5cd_w0&t=24s
+ */
+
+#include <cassert>            /// for assert
+#include <iostream>           /// for io operations
+#include <limits>             /// for variable INF
+#include <queue>              /// for the priority_queue of distances
+#include <utility>            /// for make_pair function
+#include <vector>             /// for store the graph, the distances, and the path
+
+constexpr int64_t INF = std::numeric_limits<int64_t>::max();
+
+/**
+ * @namespace graph
+ * @brief Graph Algorithms
+ */
+namespace graph {
+/**
+ * @namespace bidirectional_dijkstra
+ * @brief Functions for [Bidirectional Dijkstra Shortest Path]
+ * (https://www.coursera.org/learn/algorithms-on-graphs/lecture/7ml18/bidirectional-dijkstra) algorithm
+ */
+namespace bidirectional_dijkstra {
+/**
+ * @brief Function that add edge between two nodes or vertices of graph
+ *
+ * @param adj1 adjacency list for the direct search
+ * @param adj2 adjacency list for the reverse search
+ * @param u any node or vertex of graph
+ * @param v any node or vertex of graph
+ */
+void addEdge(std::vector<std::vector<std::pair<int, int>>> *adj1,
```

Comment:
Consider using `uint64_t` or `int64_t`, or their appropriate size ([reference](https://en.cppreference.com/w/cpp/types/integer)). Check other parts of the code.